xl/xenops_server_xenlight.ml

```diff
@@ -2995,9 +3021,11 @@ let process_one_watch xc xs (path, token) =
 	if path = _introduceDomain || path = _releaseDomain
 	then look_for_different_domains xc xs
 	else match List.filter (fun x -> x <> "") (Xstringext.String.split '/' path) with
-		| "local" :: "domain" :: domid :: "backend" :: kind :: frontend :: devid :: _ ->
+		| "local" :: "domain" :: domid :: "backend" :: kind :: frontend :: devid :: key ->
 			debug "Watch on backend domid: %s kind: %s -> frontend domid: %s devid: %s" domid kind frontend devid;
-			fire_event_on_device frontend kind devid
+			fire_event_on_device frontend kind devid;
+			(* If this event was a state change then this might be the first time we see evidence of PV drivers *)
+			if key = ["state"] then maybe_update_pv_drivers_detected ~xc ~xs (int_of_string frontend) path
```

Comment:
Indeed. I think this is reasonable for a field called "pv_drivers_detected" (rather than "pv_drivers_present", for example). Also, I believe it's the best we can ever reliably do: even if we change the PV drivers to make their presence more obvious, there will always be some released PV drivers which only reveal their existence by driving the blkif/netif state machines.